### PR TITLE
Encapsulate IOLimiter, remove rocksdb dependency from sst_importer

### PR DESCRIPTION
**Pull request by brson**
###  What have you changed?

This is another step in abstracting the storage engine, re https://github.com/tikv/tikv/issues/4184.

This builds on https://github.com/tikv/tikv/pull/5790, so most of the commits are in common. New commits begin at "Add IOLimiter, IOLimiterExt traits".

It adds an IOLimiter trait to engine_traits, and an IOLimiterExt trait to tie it to the engine. It moves the implementation from the engine crate to engine_rocks, and moves LimitReader and LimitWriter to engine_traits.

After this PR, the sst_importer crate does not depend on a concrete engine implementation, except in development mode, where it indirectly relies on RocksDB through test_sst_importer.

Ultimately IOLimiter is something that seems like it could be engine-independent, but right now it is implemented inside RocksDB, so it is part of the engine abstraction.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

cargo check, cargo test --all --no-run

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

no

###  Does this PR affect `tidb-ansible`?

no

###  Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/issues/4184

### CODE REVIEW

**Comment by Connor1996** [comment]
Please fix the conflicts

**Comment by brson** [comment]
This is rebased onto master.

File: components/engine_traits/src/io_limiter.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+pub trait IOLimiter {
+    /// # Arguments
+    ///
+    /// - `bytes_per_sec`: controls the total write rate of compaction and flush in bytes per second.
+    fn new(bytes_per_sec: u64) -> Self;
+
+    /// Sets the rate limit in bytes per second
+    fn set_bytes_per_second(&self, bytes_per_sec: i64);
```
[the remark this answers is not in the document]
Comment (brson) [reply components/engine_traits/src/io_limiter.rs]: I've made them use i64, per the underlying implementation.

**Comment by brson** [comment]
New comments addressed.

**Comment by brson** [comment]
Conflicts resolved. This is ready to merge.

**Comment by brson** [comment]
PTAL @Connor1996 @zhangjinpeng1987 @kennytm @5kbpers

**Pull request merged**